### PR TITLE
fix(crypto): use zero auxiliary data in BIP340

### PR DESCRIPTION
**Pull request by onvej-sl**
This PR:
  1. makes `zkp_bip340_sign_digest` use 32 zero bytes as auxiliary data if no auxiliary data is provided (see the rationale below)
  2. fixes tests accordingly

### Rationale
[BIP-340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#default-signing) says auxiliary data is a 32-byte array, which implicitly means the data cannot be empty. The function `secp256k1_schnorrsig_sign` from `secp256k1-zkp` incorrectly handles the case when it is provided with `NULL` as auxiliary data. The function, in that case, uses an empty array which is not in accordance with the BIP but it is completely OK from a security perspective.

The first solution that comes to mind is to use 32 zero bytes as auxiliary data if no auxiliary data is provided. This is what [NBitcoin](https://github.com/MetacoSA/NBitcoin) library does and [secp256k1-zkp](https://github.com/ElementsProject/secp256k1-zkp) does since [5324f89](https://github.com/ElementsProject/secp256k1-zkp/commit/5324f8942dd322448fae6c9b225ecac2854fa7e2).

Once we bump secp256k1-zkp submodule to the latest master, (1) can be removed from this PR.

Note that the [P2TR SLIP-19 test vector](https://github.com/satoshilabs/slips/blob/master/slip-0019.md#test-vector-5-p2tr) should be updated too.

### CODE REVIEW

**Comment by onvej-sl** [comment]
> Maybe let's just directly update the secp256k1-zkp submodule in this PR?

I've already started working on it, since I need it to finish https://github.com/trezor/trezor-firmware/issues/2085, but I encountered minor problems. It'll be a part of of this PR or https://github.com/trezor/trezor-firmware/issues/2085.

File: tests/device_tests/bitcoin/test_fujicoin.py [hunk 1]
```diff
@@ -45,5 +45,5 @@ def test_send_p2tr(client):
     # https://explorer.fujicoin.org/tx/a1c6a81f5e8023b17e6e3e51e2596d5b5e1d4914ea13c0c31cef90b3c3edee86
```
Comment (andrewkozlik) [inline tests/device_tests/bitcoin/test_fujicoin.py:45]: Obviously not valid anymore. At least add a comment to explain.

File: tests/device_tests/bitcoin/test_grs.py [hunk 2]
```diff
@@ -240,5 +240,5 @@ def test_send_p2tr(client):
     # https://blockbook-test.groestlcoin.org/tx/c66a79075044aaab3dba17daffb23f48addee87d7c87c7bc88e2997ce38a74ee
```
Comment (andrewkozlik) [inline tests/device_tests/bitcoin/test_grs.py:240]: Same comment as for Fujicoin.

File: tests/device_tests/bitcoin/test_fujicoin.py (hunk 1 above)
Comment (andrewkozlik) [reply tests/device_tests/bitcoin/test_fujicoin.py:45]: The same also applies to the BTC P2TR tests, a lot of which are on the blockchain, but will now have a different signature. Evidently `test_p2tr_with_proof` and `test_p2tr_fee_bump`, but also for example `test_send_mixed`, which is 37aa642d3cc0e0993c32e0213c92231841d3a45fcf0d03e7057ace75dd786146, and probably the others too.

**Comment by prusnak** [comment]
secp256k1-zkp has been updated in https://github.com/trezor/trezor-firmware/pull/2261

do we still need this PR?

**Comment by onvej-sl** [comment]
This draft PR was replaced by https://github.com/trezor/trezor-firmware/pull/2175 (see https://github.com/trezor/trezor-firmware/pull/2175/commits/58d64a2ca963a3f84c903367a08e2cb7bd875f87).

**Pull request closed without merging**